From 719735fd721dffe278a5336cc07dc90a53725329 Mon Sep 17 00:00:00 2001
From: Christopher James Halse Rogers <raof@ubuntu.com>
Date: Mon, 18 Sep 2023 17:54:01 +1000
Subject: [PATCH 4/4] Update for Mir 2.15 removals.

Mir 2.15 dropped some deprecated functions:
* `handle_request_drag_and_drop` is from mirclient's ContentHub interface; no longer used
* `mir_keyboard_event_key_code` was renamed to the clearer `mir_keyboard_event_keysym` (it returns keysyms)
---
 include/qtmir/windowmanagementpolicy.h                 |  1 -
 src/common/debughelpers.cpp                            |  2 +-
 src/platforms/mirserver/qteventfeeder.cpp              |  2 +-
 .../mirserver/wrappedwindowmanagementpolicy.cpp        | 10 ----------
 .../mirserver/wrappedwindowmanagementpolicy.h          |  1 -
 5 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/include/qtmir/windowmanagementpolicy.h b/include/qtmir/windowmanagementpolicy.h
index c1e5519f..f8a4c7b1 100644
--- a/include/qtmir/windowmanagementpolicy.h
+++ b/include/qtmir/windowmanagementpolicy.h
@@ -82,7 +82,6 @@ public:
     void advise_delete_window(const miral::WindowInfo &windowInfo) override;
     void advise_raise(const std::vector<miral::Window> &windows) override;
 
-    void handle_request_drag_and_drop(miral::WindowInfo &window_info) override;
     void handle_request_move(miral::WindowInfo &window_info, const MirInputEvent *input_event) override;
     void handle_request_resize(miral::WindowInfo &window_info, const MirInputEvent *input_event, MirResizeEdge edge) override;
  
diff --git a/src/common/debughelpers.cpp b/src/common/debughelpers.cpp
index 362d252d..e6245411 100644
--- a/src/common/debughelpers.cpp
+++ b/src/common/debughelpers.cpp
@@ -315,7 +315,7 @@ QString mirKeyboardEventToString(MirKeyboardEvent const* event)
 {
     MirKeyboardAction keyboardAction = mir_keyboard_event_action(event);
 
-    xkb_keysym_t keyCode = mir_keyboard_event_key_code(event);
+    xkb_keysym_t keyCode = mir_keyboard_event_keysym(event);
 
     MirInputEventModifiers modifiers = mir_keyboard_event_modifiers(event);
 
diff --git a/src/platforms/mirserver/qteventfeeder.cpp b/src/platforms/mirserver/qteventfeeder.cpp
index a385c0f5..441a7a64 100644
--- a/src/platforms/mirserver/qteventfeeder.cpp
+++ b/src/platforms/mirserver/qteventfeeder.cpp
@@ -592,7 +592,7 @@ void QtEventFeeder::dispatchKey(const MirKeyboardEvent *kev)
                 std::chrono::nanoseconds(mir_input_event_get_event_time(iev)));
     EventBuilder::instance()->store(iev, timestamp.count());
 
-    xkb_keysym_t xk_sym = mir_keyboard_event_key_code(kev);
+    xkb_keysym_t xk_sym = mir_keyboard_event_keysym(kev);
 
     // Key modifier and unicode index mapping.
     auto modifiers = getQtModifiersFromMir(mir_keyboard_event_modifiers(kev));
diff --git a/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp b/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
index 730e0dfd..2e4552b7 100644
--- a/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
+++ b/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
@@ -456,11 +456,6 @@ namespace qtmir
         return toMirRectangle(geom.marginsRemoved(windowMargins));
     }
 
-    void WindowManagementPolicy::handle_request_drag_and_drop(miral::WindowInfo &window_info)
-    {
-        MinimalWindowManager::handle_request_drag_and_drop(window_info);
-    }
-
     void WindowManagementPolicy::handle_request_move(miral::WindowInfo &window_info,
                                                     const MirInputEvent *input_event)
     {
@@ -794,11 +789,6 @@ void WrappedWindowManagementPolicy::set_window_margins(MirWindowType windowType,
     // TODO: update window positions/sizes to respect new margins.
 }
 
-void WrappedWindowManagementPolicy::handle_request_drag_and_drop(miral::WindowInfo &window_info)
-{
-    m_wrapper->handle_request_drag_and_drop(window_info);
-}
-
 void WrappedWindowManagementPolicy::handle_request_move(miral::WindowInfo &window_info,
                                                 const MirInputEvent *input_event)
 {
diff --git a/src/platforms/mirserver/wrappedwindowmanagementpolicy.h b/src/platforms/mirserver/wrappedwindowmanagementpolicy.h
index 52374ca1..59abbb00 100644
--- a/src/platforms/mirserver/wrappedwindowmanagementpolicy.h
+++ b/src/platforms/mirserver/wrappedwindowmanagementpolicy.h
@@ -77,7 +77,6 @@ public:
     void advise_delete_window(const miral::WindowInfo &windowInfo) override;
     void advise_raise(const std::vector<miral::Window> &windows) override;
 
-    void handle_request_drag_and_drop(miral::WindowInfo &window_info) override;
     void handle_request_move(miral::WindowInfo &window_info, const MirInputEvent *input_event) override;
     void handle_request_resize(miral::WindowInfo &window_info, const MirInputEvent *input_event, MirResizeEdge edge) override;
 
-- 
2.43.0

